src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -442,9 +443,16 @@ else if (planKey.toLowerCase().contains("-solution")) { // TODO: transfer this i
         }
 
         try {
-            List<Feedback> feedbackItems = feedbackService.getFeedbackForBuildResult(result.get());
-            // TODO: send an empty list to the client and do not send a 404
-            return Optional.ofNullable(feedbackItems).map(resultDetails -> new ResponseEntity<>(feedbackItems, HttpStatus.OK)).orElse(notFound());
+            // There is a difference between an empty list of feedback items & no feedback items at all.
```

Comment:
readd this call again for legacy exercises (as discussed this can be removed in the September 2019)

Reply:
Restored.